app/interactors/oauth_organizer.rb

```diff
@@ -0,0 +1,44 @@
+class OauthOrganizer
+  class OauthError < StandardError
+  end
+
+  attr_reader :auth, :current_user
+  private :auth, :current_user
+
+  def initialize(current_user, auth)
+    @current_user = current_user
+    @auth = auth
+  end
+
+  def call
+    user.present? ? connect_social_profile : fail_oauth
+    user
+  end
+
+  private
+
+  def user
+    @user ||= current_user || fetch_oauth_user || build_user
+  end
+
+  def fetch_oauth_user
+    FetchOauthUser.new(auth).call if auth_verified?
+  end
+
+  def auth_verified?
+    AuthVerificationPolicy.new(auth).verified?
+  end
+
+  def build_user
+    User.build_from_omniauth(auth) if auth_verified?
```

Comment:
Hm, I doubt that we should build something from unverified auth, so auth verification should be kept